File: backend/pawnscan/src/main/java/edu/cit/garbo/pawnscan/dto/AuthResponse.java
package edu.cit.garbo.pawnscan.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponse {

    private Long userId;
    private String email;
    private String fullName;
    private String role;
    private String token;
    private BusinessProfileSummaryResponse businessProfile;
    private String message;
}

File: mobile/app/src/main/java/com/cit/pawnscan/api/AuthResponse.kt
package com.cit.pawnscan.api

import com.google.gson.annotations.SerializedName

data class AuthResponse(
    @SerializedName("userId")
    val userId: Long? = null,

    @SerializedName("email")
    val email: String? = null,

    @SerializedName("fullName")
    val fullName: String? = null,

    @SerializedName("role")
    val role: String? = null,

    @SerializedName("token")
    val token: String? = null,

    @SerializedName("businessProfile")
    val businessProfile: BusinessProfileSummary? = null,

    @SerializedName("message")
    val message: String? = null
)

data class BusinessProfileSummary(
    @SerializedName("id")
    val id: Long? = null,

    @SerializedName("businessName")
    val businessName: String? = null,

    @SerializedName("businessAddress")
    val businessAddress: String? = null
)
